domain/status/state/crossmodelrelation.go

```diff
@@ -0,0 +1,211 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair"
+
+	coreremoteapplication "github.com/juju/juju/core/remoteapplication"
+	applicationerrors "github.com/juju/juju/domain/application/errors"
+	crossmodelrelationerrors "github.com/juju/juju/domain/crossmodelrelation/errors"
+	domainlife "github.com/juju/juju/domain/life"
+	"github.com/juju/juju/domain/status"
+	statuserrors "github.com/juju/juju/domain/status/errors"
+	internaldatabase "github.com/juju/juju/internal/database"
+	"github.com/juju/juju/internal/errors"
+)
+
+// GetRemoteApplicationOffererUUIDByName returns the UUID for the named for the remote
+// application wrapping the named application
+func (st *ModelState) GetRemoteApplicationOffererUUIDByName(
+	ctx context.Context,
+	name string,
+) (coreremoteapplication.UUID, error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return "", errors.Capture(err)
+	}
+
+	var uuid coreremoteapplication.UUID
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		var err error
+		uuid, err = st.lookupRemoteApplicationOfferer(ctx, tx, name)
+		return err
+	}); err != nil {
+		return "", errors.Capture(err)
+	}
+
+	return uuid, nil
+}
+
+// GetRemoteApplicationOffererStatus returns the status of the specified remote
+// application in the local model.
+func (st *ModelState) GetRemoteApplicationOffererStatus(
+	ctx context.Context,
+	uuid string,
+) (status.StatusInfo[status.WorkloadStatusType], error) {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return status.StatusInfo[status.WorkloadStatusType]{}, errors.Capture(err)
+	}
+
+	ident := remoteApplicationUUID{RemoteApplicationUUID: uuid}
+
+	query, err := st.Prepare(`
+SELECT &statusInfo.*
+FROM application_remote_offerer_status
+WHERE application_remote_offerer_uuid = $remoteApplicationUUID.uuid
```

Comment:
Here you could just `JOIN application ` and receive the appName instead of 2 db reads.